test/test_modules.py

```diff
@@ -396,6 +397,11 @@ def _test_gradients_helper(self, device, dtype, module_info, check):
         module_cls = module_info.module_cls
         module_inputs = module_info.module_inputs_func(module_info, device=device, dtype=dtype,
                                                        requires_grad=True)
+        # Set some backend-specific validation settings.
+        gradcheck_nondet_tol = 0.0
+        if torch.backends.cudnn.is_available():
+            # cuDNN introduces non-determinism
+            gradcheck_nondet_tol = GRADCHECK_NONDET_TOL
```

Comment:
After thinking about this some more, I realized it's probably better to set `gradcheck_nondet_tol` on a per-module basis instead of always setting it when cudnn is enabled.

The reason for this is because setting it like this is possibly too liberal, and wouldn't catch certain cases that we do want to catch involving non-determinism. I think the ideal state would be to set the tolerance on a per-module basis only for CUDA tests if cuDNN is enabled, if there's some way to accomplish that. Then we are as strict as possible with the tolerance and only increase it when absolutely necessary. Wdyt about this?